### PR TITLE
imx6ull-ecspi: Added library for interfacing with SPI in i.MX

### PR DESCRIPTION
**Pull request by Darchiv**

### CODE REVIEW

File: spi/imx6ull-ecspi/ecspi-api.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL ECSPI lib API
+ *
+ * Copyright 2018 Phoenix Systems
+ * Author: Daniel Sawka, Krystian Wasik
+ *
+ * This file is part of Phoenix-RTOS.
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-api.h]: This file is not part of Phoenix-RTOS, please remove this line.

File: spi/imx6ull-ecspi/ecspi-lib.c [hunk 2]
```diff
@@ -0,0 +1,303 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL ECSPI lib
+ *
+ * Copyright 2018 Phoenix Systems
+ * Author: Daniel Sawka, Krystian Wasik
+ *
+ * This file is part of Phoenix-RTOS.
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-lib.c]: This file is not part of Phoenix-RTOS, please remove this line.

File: spi/imx6ull-ecspi/ecspi-lib.c [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+}
+
+static void set_clk(int dev_no)
+{
+	platformctl_t ctl;
+
+	ctl.action = pctl_set;
+	ctl.type = pctl_devclock;
+	ctl.devclock.dev = ecspi_pctl_clk[dev_no - 1];
+	ctl.devclock.state = 0b11;
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-lib.c]: Please do not use binary numbers.

File: spi/imx6ull-ecspi/ecspi-lib.c [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+	if (burst % 32 == 1) {
+		printf("ecspi: burst length of 32n + 1 is erroneous in hardware itself\n");
+		return;
+	}
+
+	e = &ecspi[dev_no - 1];
+	*(e->base + conreg) = (*(e->base + conreg) & ~(0xFFF << 20)) | ((burst - 1) << 20);
+}
+
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-lib.c]: Please keep two empty lines distance between functions bodies.

File: spi/imx6ull-ecspi/ecspi-lib.c [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+			word = (word << 8) | *out++;
+			fill_len--;
+			len--;
+		}
+
+		*(e->base + txdata) = word;
+	}
+
+	while (len > 0) {
+		word = out[3] | (out[2] << 8) | (out[1] << 16) | (out[0] << 24);
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-lib.c]: Maybe cast _out_ to uint32_t?

File: spi/imx6ull-ecspi/ecspi-lib.c [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+{
+	ecspi_t *e;
+
+	if (dev_no < 1 || dev_no > 4) {
+		return -1;
+	}
+
+	/* Default clock for ECSPI is 60 MHz. */
+	e = &ecspi[dev_no - 1];
+	*(e->base + conreg) = (*(e->base + conreg) & ~(0b11111111 << 8)) | ((pre & 0x0F) << 12) | ((post & 0x0F) << 8);
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/ecspi-lib.c]: Please do not use binary numbers.

File: spi/imx6ull-ecspi/imx6ull-ecspi.c [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL ECSPI driver
+ *
+ * Copyright 2018 Phoenix Systems
+ * Author: Daniel Sawka
+ *
+ * This file is part of Phoenix-RTOS.
```
Comment (agkaminski) [inline spi/imx6ull-ecspi/imx6ull-ecspi.c]: This file is not part of Phoenix-RTOS, please remove this line.

File: spi/imx6ull-ecspi/ecspi-lib.c (hunk 5 above)
Comment (Darchiv) [reply spi/imx6ull-ecspi/ecspi-lib.c]: Casting will reorder bytes which is unacceptable.

**Pull request merged**